stream_alert_cli/terraform/lambda_module.py

```diff
@@ -0,0 +1,128 @@
+
+        result['{}_alarm_enabled'.format(alarm_type)] = True
+        result['{}_alarm_evaluation_periods'.format(alarm_type)] = settings.get(
+            'evaluation_periods', 1)
+        result['{}_alarm_period_secs'.format(alarm_type)] = settings.get('period_secs', 120)
+        result['{}_alarm_threshold'.format(alarm_type)] = settings.get('threshold', 0)
+
+    return result
+
+
+def generate_lambda(function_name, config, environment=None):
+    """Generate an instance of the Lambda Terraform module.
+
+    Args:
+        function_name (str): Name of the Lambda function (e.g. 'alert_processor')
+        config (dict): Parsed config from conf/
+        environment (dict): Optional environment variables to specify.
+            LOGGER_LEVEL is included automatically.
+
+    Example Lambda config:
+        {
+            "concurrency_limit": 1,
+            "current_version": "$LATEST",
+            "handler": "main.handler",
+            "invocation_frequency_minutes": 5,
+            "log_level": "info",
+            "log_retention_days": 14,
+            "memory": 128,
+            "metric_alarms": {
+              "errors": {
+                "enabled": true,
+                "evaluation_periods": 1,
+                "period_secs": 120,
+                "threshold": 0
+              },
+              "throttles": {
+                "enabled": true,
+                "evaluation_periods": 1,
+                "period_secs": 120,
+                "threshold": 0
+              }
+            },
+            "source_bucket": "BUCKET",
+            "source_object_key": "OBJECT_KEY",
+            "timeout": 10,
+            "vpc_config": {
+                "security_group_ids": [
+                    "sg-id"
+                ],
+                "subnet_ids": [
+                    "subnet-id"
+                ]
+            }
+        }
+
+    Returns:
+        dict: Terraform config for an instance of the tf_lambda module.
+    """
+    lambda_config = _lambda_config(function_name, config)
+    vpc_config = lambda_config.get('vpc_config', {})
+
+    # Add logger level to any custom environment variables
+    environment_variables = {
+        'LOGGER_LEVEL': lambda_config.get('log_level', 'info')
+    }
+    if environment:
+        environment_variables.update(environment)
+
+    lambda_module = {
+        'source': 'modules/tf_lambda',
+        'function_name': '{}_streamalert_{}'.format(config['global']['account']['prefix'],
+                                                    function_name),
+        'description': 'StreamAlert {}'.format(function_name.replace('_', ' ').title()),
+        'handler': lambda_config['handler'],
+        'memory_size_mb': lambda_config['memory'],
+        'timeout_sec': lambda_config['timeout'],
+        'source_bucket': lambda_config['source_bucket'],
+        'source_object_key': lambda_config['source_object_key'],
+        'concurrency_limit': lambda_config.get('concurrency_limit', ''),
```

Comment:
Isn't the module default `''`? Should we declare it in both places?

Reply:
This is one of the negative side effects of marshaling our custom config to the Terraform variables - there is a lot of duplication. (I'm exploring other ways to do this.)

In the meantime, you're right - our general strategy has been not to marshal data which is not present in the config (which will then use the module default). I've updated the Lambda module generation logic to follow the same pattern (at the cost of some complexity)